### PR TITLE
docs: add requireSync note, ssr root route, app generation, and custom URL prefix

### PR DESCRIPTION
**Pull request by leblancmeneses**
docs(content): requireSync note, ssr root route, app generation, and custom URL prefix

### CODE REVIEW

File: apps/docs-app/docs/features/data-fetching/server-side-data-fetching.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ export const load = async ({
 ## Injecting the Data
 
 Accessing the data fetched on the server can be done using the `injectLoad` function provided by `@analogjs/router`.
+The `load` function here is resolved using Angular route resolvers so `requireSync: false, initialValue: {}` provide no benefit
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/data-fetching/server-side-data-fetching.md:29]:
```suggestion
The `load` function here is resolved using Angular route resolvers so `requireSync: false, initialValue: {}` has no benefit.
```

File: apps/docs-app/docs/features/deployment/overview.md [hunk 2]
```diff
@@ -51,37 +51,33 @@ export default defineConfig({
...
-If you are deploying with a custom URL prefix, such as https://domain.com/`basehref`/ you must do these steps for [server-side-data-fetching](https://analogjs.org/docs/features/data-fetching/server-side-data-fetching), [html markup and assets](https://angular.io/api/common/APP_BASE_HREF), and [dynamic api routes](https://analogjs.org/docs/features/api/overview) to work correctly on the specified `basehref`.
+If you are deploying with a custom URL prefix, such as https://domain.com/ `basehref` you must do these steps for [server-side-data-fetching](https://analogjs.org/docs/features/data-fetching/server-side-data-fetching), [html markup and assets](https://angular.io/api/common/APP_BASE_HREF), and [dynamic api routes](https://analogjs.org/docs/features/api/overview) to work correctly on the specified `basehref`.
 
-1. Instruct Angular on how recognize and generate URLs. Create a new file `app.config.env.ts`.
+1. Update the `app.config.ts` file to use the new file.
+
+This instructs Angular on how recognize and generate URLs.
 
 ```ts
-import { ApplicationConfig } from '@angular/core';
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/deployment/overview.md:59]: Why was this removed?

File: apps/docs-app/docs/features/deployment/overview.md [hunk 3]
```diff
@@ -95,7 +91,20 @@ Given a `vite.config.ts` file similar to this:
 ```ts
     plugins: [
       analog({
+        ssr: true,
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/deployment/overview.md:94]: This is the default

```suggestion
```

File: apps/docs-app/docs/features/server/server-side-rendering.md [hunk 4]
```diff
@@ -64,4 +64,30 @@ export default defineConfig(({ mode }) => ({
...
+      nitro: {
+        routeRules: {
+          '/': {
+            prerender: false,
+          },
+        },
+      },
+      prerender: {
+        routes: async () => {
+          return ['/'];
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/server/server-side-rendering.md:87]:
```suggestion
          return [];
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: apps/docs-app/docs/features/deployment/overview.md (hunk 2 above)
Comment (leblancmeneses) [reply apps/docs-app/docs/features/deployment/overview.md:59]: I got rid of app.config.env.ts with mergeApplicationConfig in favor of using import.meta.env.BASE_URL

File: apps/docs-app/docs/features/server/server-side-rendering.md (hunk 4 above)
Comment (leblancmeneses) [reply apps/docs-app/docs/features/server/server-side-rendering.md]: Done.  FYI: without ['/'] the root url is still not server side rendered. I just tried it out to make sure.

File: apps/docs-app/docs/features/deployment/overview.md (hunk 2 above)
Comment (brandonroberts) [reply apps/docs-app/docs/features/deployment/overview.md:59]: Yes, but `ApplicationConfig` is still being used in the code snippet

Comment (leblancmeneses) [reply apps/docs-app/docs/features/deployment/overview.md:59]: Got it! Done.

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks! @allcontributors add @leblancmeneses for doc
